lib/pbench/cli/server/report.py

```diff
@@ -0,0 +1,403 @@
+            try:
+                referenced = tarball.last_ref.stat().st_mtime
+            except Exception as e:
+                detailer.write(f"{tarball.name} last ref access: {str(e)!r}")
+                last_ref_errors += 1
+            else:
+                if not oldest_cache or referenced < oldest_cache:
+                    oldest_cache = referenced
+                    oldest_cache_name = tarball.name
+                if not newest_cache or referenced > newest_cache:
+                    newest_cache = referenced
+                    newest_cache_name = tarball.name
+            cached_count += 1
+            size = Metadata.getvalue(tarball.dataset, Metadata.SERVER_UNPACKED)
+            if not size:
+                detailer.write(f"{tarball.name} has no unpacked size")
+                lacks_size += 1
+            elif not isinstance(size, int):
+                detailer.write(
+                    f"{tarball.name} has non-integer unpacked size "
+                    f"{size!r} ({type(size)})"
+                )
+                bad_size += 1
+            else:
+                if not smallest_cache or size < smallest_cache:
+                    smallest_cache = size
+                    smallest_cache_name = tarball.name
+                if not biggest_cache or size > biggest_cache:
+                    biggest_cache = size
+                    biggest_cache_name = tarball.name
+                cached_size += size
+    oldest = datetime.datetime.fromtimestamp(oldest_cache, datetime.timezone.utc)
+    newest = datetime.datetime.fromtimestamp(newest_cache, datetime.timezone.utc)
+    click.echo("Cache report:")
+    click.echo(
+        f"  {cached_count} datasets are cached, consuming "
+        f"{humanize.naturalsize(cached_size)}"
+    )
+    click.echo(
+        f"  {lacks_size} datasets have never been unpacked, "
+        f"{last_ref_errors} are missing reference timestamps, "
+        f"{bad_size} have bad size metadata"
+    )
+    click.echo(
+        f"  The smallest cache, {smallest_cache_name}, is "
+        f"{humanize.naturalsize(smallest_cache)}"
+    )
+    click.echo(
+        f"  The biggest cache, {biggest_cache_name}, is "
+        f"{humanize.naturalsize(biggest_cache)}"
+    )
+    click.echo(
+        f"  The least recently used cache, {oldest_cache_name}, was "
+        f"referenced {humanize.naturaldate(oldest)}"
+    )
+    click.echo(
+        f"  The most recently used cache, {newest_cache_name}, was "
+        f"referenced {humanize.naturaldate(newest)}"
+    )
+
+
+def report_sql():
+    """Report the SQL table storage statistics"""
+    click.echo("SQL storage report:")
+    click.echo("  Table                Rows       Storage")
+    click.echo("  -------------------- ---------- ----------")
+    for t in inspect(Database.db_session.get_bind()).get_table_names():
+        rows = list(
+            Database.db_session.execute(statement=text(f"SELECT COUNT(*) FROM {t}"))
+        )[0][0]
```

Comment:
I think you can use `next()` instead of `list()`:
```python
        rows = next(
            Database.db_session.execute(statement=text(f"SELECT COUNT(*) FROM {t}"))
        )[0]
```
Ditto line 231.

Reply:
Dang. I knew there was a way to do that, but I forgot the syntax and a quick search or two didn't reveal it.